src/Symfony/Component/HttpKernel/DependencyInjection/RegisterControllerArgumentLocatorsPass.php

```diff
@@ -136,14 +136,17 @@ public function process(ContainerBuilder $container)
                         $binding = $bindings[$bindingName];
 
                         list($bindingValue, $bindingId) = $binding->getValues();
+                        $binding->setValues(array($bindingValue, $bindingId, true));
 
                         if (!$bindingValue instanceof Reference) {
-                            continue;
+                            $args[$p->name] = new Reference('value.'.$container->hash($bindingValue));
+                            $container->register((string) $args[$p->name], 'mixed')
+                                ->setFactory('current')
+                                ->addArgument(array($bindingValue));
```

Comment:
As this service is here to build a scalar, I should we should mark it as non-shared, to avoiding registering the scalar in `$this->services`

Reply:
why? what's the issue with sharing a scalar (or array btw?)
looks better shared to me, did I miss something?